client/tests/integration/permissions.rs

```diff
@@ -293,3 +285,54 @@ fn permissions_differ_not_only_by_names() {
         .submit_blocking(set_shoes_color)
         .expect("Failed to modify Mouse's shoes");
 }
+
+#[test]
+fn stored_vs_granted_token_payload() -> Result<()> {
+    let (_rt, _peer, iroha_client) = <PeerBuilder>::new().with_port(10_730).start_with_runtime();
+    wait_for_genesis_committed(&[iroha_client.clone()], 0);
+
+    // Given
+    let alice_id = AccountId::from_str("alice@wonderland").expect("Valid");
+
+    // Registering mouse and asset definition
+    let asset_definition_id: AssetDefinitionId = "xor#wonderland".parse().expect("Valid");
+    let create_asset = RegisterBox::new(AssetDefinition::store(asset_definition_id.clone()));
+    let mouse_id: AccountId = "mouse@wonderland".parse().expect("Valid");
+    let mouse_keypair = iroha_crypto::KeyPair::generate().expect("Failed to generate KeyPair.");
+    let new_mouse_account = Account::new(mouse_id.clone(), [mouse_keypair.public_key().clone()]);
+    let instructions: [InstructionBox; 2] = [
+        RegisterBox::new(new_mouse_account).into(),
+        create_asset.into(),
+    ];
+    iroha_client
+        .submit_all_blocking(instructions)
+        .expect("Failed to register mouse");
+
+    // Allow alice to mint mouse asset and mint initial value
+    let mouse_asset = AssetId::new(asset_definition_id, mouse_id.clone());
+    let allow_alice_to_set_key_value_in_mouse_asset = GrantBox::new(
+        PermissionToken::from_str_unchecked(
+            "CanSetKeyValueInUserAsset".parse().unwrap(),
+            // NOTE: Introduced additional whitespaces in the serialized form
+            "{ \"asset_id\" : \"xor#wonderland#mouse@wonderland\" }",
+        ),
+        alice_id,
+    );
+
+    let transaction = TransactionBuilder::new(mouse_id)
+        .with_instructions([allow_alice_to_set_key_value_in_mouse_asset])
+        .sign(mouse_keypair)
+        .expect("Failed to sign mouse transaction");
```

Comment:
Not a suitable message here.

```suggestion
        .expect("Should sign mouse transaction fine");
```

```suggestion
        .unwrap();
```